src/Compilers/VisualBasic/Portable/Symbols/ReferenceManager.vb

```diff
@@ -361,9 +359,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 End If
 
                 ' Setup references for the compilation
-                referencedAssembliesMap = New Dictionary(Of MetadataReference, ReferencedAssembly)(referenceMap.Length)
+                referencedAssembliesMap = New Dictionary(Of MetadataReference, Integer)(referenceMap.Length)
                 referencedModulesMap = New Dictionary(Of MetadataReference, Integer)(modules.Length)
 
+                Dim referencedAssemblies = ArrayBuilder(Of AssemblySymbol).GetInstance()
+                Dim aliasesOfReferencedAssemblies = ArrayBuilder(Of ImmutableArray(Of String)).GetInstance()
+
                 Dim sourceModule = assemblySymbol.SourceModule
                 Dim referencedAssemblySymbols = sourceModule.GetReferencedAssemblySymbols()
```

Comment:
Yes, I convinced myself that these are indeed the same arrays and we don't need to rebuild the array. I also run tests with appropriate asserts that verify this and all pass.